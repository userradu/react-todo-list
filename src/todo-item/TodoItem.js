import React from 'react';
import './TodoItem.css';

function TodoItem(props) {
	function handleTodoItemStatusModified(event) {
		props.onTodoItemStatusModified({
			id: props.todoItem.id,
			status: event.target.checked ? 'Complete' : 'Incomplete'
		});
	}

	return (
		<div>
      <input
        type="checkbox"
        checked={props.todoItem.status === 'Complete'}
        onChange={handleTodoItemStatusModified} />
			<span className={props.todoItem.status === 'Complete' ? 'text-strike' : null}>
				{props.todoItem.name}
			</span>
		</div>
	);
}

export default TodoItem;
