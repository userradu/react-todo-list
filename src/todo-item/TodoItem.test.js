import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import TodoItem from './TodoItem';

describe('TodoItem', () => {

  it('renders correctly', () => {
    const todoItem = {
      id: 1,
      name: 'todo item',
      status: 'Complete'
    };

    const tree = renderer.create(
      <TodoItem
        onTodoItemStatusModified={() => {}}
        todoItem={todoItem} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('calls props.onTodoItemStatusModified() with the correct arguments when the checkbox is checked', () => {
    const onTodoItemStatusModified = sinon.spy();
    const todoItem = {
      id: 1,
      name: 'todo item',
      status: 'Complete'
    };

    const wrapper = shallow(<TodoItem
      onTodoItemStatusModified={onTodoItemStatusModified}
      todoItem={todoItem} />);

    const input = wrapper.find('input');
    input.simulate('change', {
      target: { checked: true }
    });

    expect(onTodoItemStatusModified.called).toBe(true);
    expect(onTodoItemStatusModified.calledWith({
      id: todoItem.id,
      status: 'Complete'
    })).toBe(true);
  });

  it('calls props.onTodoItemStatusModified() with the correct arguments when the checkbox is unchecked', () => {
    const onTodoItemStatusModified = sinon.spy();
    const todoItem = {
      id: 1,
      name: 'todo item',
      status: 'Complete'
    };

    const wrapper = shallow(<TodoItem
      onTodoItemStatusModified={onTodoItemStatusModified}
      todoItem={todoItem} />);

    const input = wrapper.find('input');
    input.simulate('change', {
      target: { checked: false }
    });

    expect(onTodoItemStatusModified.called).toBe(true);
    expect(onTodoItemStatusModified.calledWith({
      id: todoItem.id,
      status: 'Incomplete'
    })).toBe(true);
  });

  it('sets the checkbox as checked when the status is "Complete"', () => {
    const todoItem = {
      id: 1,
      name: 'todo item',
      status: 'Complete'
    };

    const wrapper = shallow(<TodoItem
      onTodoItemStatusModified={() => {}}
      todoItem={todoItem} />);

    const input = wrapper.find('input');
    expect(input.props().checked).toBe(true);
  });

  it('sets the checkbox as unchecked when the status is "Incomplete"', () => {
    const todoItem = {
      id: 1,
      name: 'todo item',
      status: 'Incomplete'
    };

    const wrapper = shallow(<TodoItem
      onTodoItemStatusModified={() => {}}
      todoItem={todoItem} />);

    const input = wrapper.find('input');
    expect(input.props().checked).toBe(false);
  });

});
