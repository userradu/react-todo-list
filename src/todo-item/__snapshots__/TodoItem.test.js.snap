// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem /> renders correctly 1`] = `
<div>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="text-strike"
  >
    todo item
  </span>
</div>
`;
